### PR TITLE
Fix the github CI action

### PR DESCRIPTION
**Pull request by ufoscout**

### CODE REVIEW

**Review by itsyaasir — APPROVED** [review]

File: ic-payments/test-payment-canister/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,11 @@
+pub mod canister;
+
+// use canister::PaymentCanister;
```
Comment (veeso) [inline ic-payments/test-payment-canister/src/main.rs:3]: is this still necessary 🤔

File: ic-stable-structures/src/memory_utils.rs [hunk 2]
```diff
@@ -5,7 +5,7 @@ use dfinity_stable_structures::{
     Memory,
 };
 
-#[cfg(all(not(target_arch = "wasm32")))]
+#[cfg(not(target_arch = "wasm32"))]
```
Comment (veeso) [inline ic-stable-structures/src/memory_utils.rs:8]: we've started to use target_family wasm on evm, maybe we can use it also here

File: ic-payments/tests/state_machine.rs [hunk 3]
```diff
@@ -109,6 +136,8 @@ mod tests {
         Decode!(&response, Option<Nat>).unwrap()
     }
 
+    // It fails
```
Comment (F3kilo) [inline ic-payments/tests/state_machine.rs:139]: Should we fix it?

**Review by F3kilo — APPROVED** [review]

**Review by itsyaasir — APPROVED** [review]

File: .github/workflows/build_and_tests.yml [hunk 4]
```diff
@@ -10,77 +10,32 @@ concurrency:
...
           cargo install ic-wasm
-          dfx start --background
-          dfx canister create dummy_canister
-          dfx build dummy_canister
-          ic-payments/build_payments_canister.sh
-
-      - name: Run integration tests
+      - name: rustfmt
+        run: |
+          cargo fmt --check
```
[the remark this answers is not in the document]
Comment (ufoscout) [reply .github/workflows/build_and_tests.yml:31]: done

File: ic-stable-structures/src/memory_utils.rs (hunk 2 above)
Comment (ufoscout) [reply ic-stable-structures/src/memory_utils.rs:8]: done

File: ic-payments/test-payment-canister/src/main.rs (hunk 1 above)
Comment (ufoscout) [reply ic-payments/test-payment-canister/src/main.rs:3]: removed

File: ic-payments/tests/state_machine.rs (hunk 3 above)
Comment (ufoscout) [reply ic-payments/tests/state_machine.rs:139]: maybe or maybe not. This test was not executed before this PR so I don't know if it ever worked or when it was broken. In addition, we are not using ic-payments anymore so I would not spend time with it

**Pull request merged**